synapse/http/proxyagent.py

```diff
@@ -243,28 +244,45 @@ def request(self, method, uri, headers=None, bodyProducer=None):
         )
 
 
-def _http_proxy_endpoint(proxy: Optional[bytes], reactor, **kwargs):
+def _http_proxy_endpoint(
+    proxy: Optional[bytes],
+    reactor: IReactorCore,
+    tls_options_factory: Optional[IPolicyForHTTPS],
```

Comment:
it seems like we rely on this not being None.

```suggestion
    tls_options_factory: IPolicyForHTTPS,
```